### ISSUE TITLE
Using enums for generic types

### ISSUE THREAD

**Issue by objeck** [objeck/objeck-lang#34]
The following code should generate a compiler error. The incorrect code causes the memory manager to incorrectly treat a literal as an object.

```
use Collection.Generic;

class Foo {
	enum Value {
		BAR
	}
	
	function : Main(args : String[]) ~ Nil {
		hash := Hash->New()<String, Foo->Value>;
		hash->Insert("foo bar", Foo->Value->BAR);
	}
}```

**Comment by objeck** [comment]
fixed in v5.9.5

**Comment by objeck** [comment]
fixed in v5.9.5
